### PR TITLE
[quant][fx2trt] Add lowering support for reference linear/conv modules

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #64787
* #64857
* __->__ #64368

Summary:

Test Plan:
python torch/fx/experimental/fx2trt/example/quantized_resnet_test.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30708738](https://our.internmc.facebook.com/intern/diff/D30708738)

### CODE REVIEW

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30708738).

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30708738).

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30708738).

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30708738).

File: torch/fx/experimental/fx2trt/converters/acc_ops_converters.py [hunk 1]
```diff
@@ -241,16 +241,35 @@ def acc_ops_conv2d(network, target, args, kwargs, name):
     if has_dynamic_shape(input_val.shape):
         assert input_val.shape[1] != -1, "Channel dim can't be dynamic for convolution."
 
-    kernel = to_numpy(kwargs["weight"])
+    # for now we'll assume bias is constant Tensor or None,
+    # and bias being ITensor is not supported in TensorRT api
+    # right now
     bias = to_numpy(kwargs["bias"])
```
Comment (842974287) [inline torch/fx/experimental/fx2trt/converters/acc_ops_converters.py:247]: Let's add a check in `to_numpy()` to fail loud if sees something that's not a `torch.Tensor`.

File: torch/fx/experimental/fx2trt/converters/acc_ops_converters.py [hunk 2]
```diff
@@ -241,16 +241,35 @@ def acc_ops_conv2d(network, target, args, kwargs, name):
...
 
-    layer = network.add_convolution(
-        input=input_val,
-        num_output_maps=kernel.shape[0],
-        kernel_shape=kernel.shape[2:],
-        kernel=kernel,
-        bias=bias,
-    )
+    weight = get_trt_tensor(network, kwargs["weight"], f"{name}_weight")
+    weight_shape = tuple(kwargs["weight"].shape)
```
Comment (842974287) [inline torch/fx/experimental/fx2trt/converters/acc_ops_converters.py:250]: Maybe move these two lines into the if branch below?

File: torch/fx/experimental/fx2trt/converters/acc_ops_converters.py [hunk 3]
```diff
@@ -1019,43 +1038,48 @@ def acc_ops_linear(network, target, args, kwargs, name):
         "dim for linear and it can't be the last dim."
     )
 
-    weight = kwargs["weight"]
-
     # For quantization, weight here would be a trt tensor because it goes through
     # quant + dequant. In this case, we need to use matmul + add because fully_connected
     # can't take non-constant weight.
```
Comment (842974287) [inline torch/fx/experimental/fx2trt/converters/acc_ops_converters.py:1043]: we can remove this part of the comments now.

File: torch/fx/experimental/fx2trt/converters/acc_ops_converters.py [hunk 4]
```diff
@@ -1307,7 +1331,9 @@ def acc_ops_permute(network, target, args, kwargs, name):
 
 @tensorrt_converter(acc_ops.quantize_per_tensor)
 def acc_ops_quantize_per_tensor(network, target, args, kwargs, name):
-    input_val = kwargs["input"]
+    # input_val = kwargs["input"]
```
Comment (842974287) [inline torch/fx/experimental/fx2trt/converters/acc_ops_converters.py:1334]: Remove

**Review by 842974287 — APPROVED** [review]
Looks great! Please also check internal CI to make sure all tests passed since right now all fx2trt tests lives in internal CI.

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30708738).

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30708738).

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30708738).

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30708738).

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30708738).

**Pull request closed without merging**